Cassette/CASQueueFile.m

```diff
@@ -504,15 +504,21 @@ - (NSUInteger)usedBytes {
  * Stores a 32-bit integer @c value in the @c buffer at the given @c offset.
  */
 void writeInt(NSMutableData *buffer, NSUInteger offset, uint32_t value) {
-    [buffer replaceBytesInRange:NSMakeRange(offset, 4) withBytes:&value];
+    BOOL hasValidRange = buffer.length >= offset + 4;
+    if (hasValidRange) {
+        [buffer replaceBytesInRange:NSMakeRange(offset, 4) withBytes:&value];
```

Comment:
We need proper error handling if we're going to do this, otherwise consumers will not know why their data is not what they expect.

Reply:
we do not have enough information from JIRA to understand what core cause is on this. 
I have changes locally for adding proper error handling, but it will alter many several API methods.

Reply:
We may not have information from JIRA, but we only have a few places where we adjust the buffer and its size. 

In other words, the symptom is we are writing a value outside of the range of a buffer. The root issue is why are we providing a bad buffer in the first place? 

I think there is an issue in our file expansion code.


re: error handling -- I think that's fine. It could pollute our internal code though, so we may just want to use try-catches internally.